### ISSUE TITLE
Loss function problem

### ISSUE THREAD

**Issue by AmericaBG** [ahukui/Deeply-supervised-CNN#2]
Hi! I solved the last problem (my error was not to change one "channel_first" parameter).

Now, I've a problem with loss function. I'm using a different one to "dice_coef".

The error says "ValueError: When passing a list as loss, it should have one entry per model outputs. The model has 9 outputs, but you passed loss=[<function focal_tversky at 0x000002B302A74048>]"

Could you help me??
Thank you very much for your attention!

**Comment by ahukui** [comment]
Can you show me your code?

**Comment by AmericaBG** [comment]
Yes, of course. I use your NetWork() code as you but I finish it with: ( I don't undestand why he model have 8 outputs if the goal is to generate a binary mask)

model = Model(img_input, [output0, output1, output2, output3, output4, output5, output6, output7, output8])
  
model.compile(optimizer =opt,loss= [focal_tversky], metrics = [matthews_correlation])
return model


My loss function is: 
def focal_tversky(y_true,y_pred):
    pt_1 = tversky(y_true, y_pred)
    gamma = 1.33
    return K.pow((1-pt_1), gamma)


Thank you very much!

**Comment by ahukui** [comment]
loss= [focal_tversky]->loss= focal_tversky




------------------ 原始邮件 ------------------
发件人: "América Bueno Gómez"<notifications@github.com>;
发送时间: 2019年10月24日(星期四) 下午4:59
收件人: "ahukui/Deeply-supervised-CNN"<Deeply-supervised-CNN@noreply.github.com>;
抄送: "kui"<1548366814@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [ahukui/Deeply-supervised-CNN] Loss function problem (#2)




Yes, of course. I use your NetWork() code as you but I finish it with: ( I don't undestand why he model have 8 outputs if the goal is to generate a binary mask)
 
model = Model(img_input, [output0, output1, output2, output3, output4, output5, output6, output7, output8])
 
model.compile(optimizer =opt,loss= [focal_tversky], metrics = [matthews_correlation])
 return model
 
My loss function is:
 def focal_tversky(y_true,y_pred):
 pt_1 = tversky(y_true, y_pred)
 gamma = 1.33
 return K.pow((1-pt_1), gamma)
 
Thank you very much!
